public/index.html

```diff
@@ -0,0 +1,237 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta
+      name="description"
+      content="Generate a GitHub Actions workflow file for running MATLAB code." />
+    <meta name="theme-color" content="#f8f9fa" />
+    <link rel="icon" href="assets/favicon.ico" />
+    <title>GitHub Actions Workflow Generator for MATLAB</title>
+
+    <link
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css"
+      rel="stylesheet" />
+    <link href="style.css" rel="stylesheet" />
+  </head>
+
+  <body>
+    <noscript>
+      <div class="alert alert-warning text-center" role="alert">
+        This site requires JavaScript to function. Please enable JavaScript in
+        your browser.
+      </div>
+    </noscript>
+
+    <main class="container py-5">
+      <div class="text-center mb-4">
+        <img
+          src="assets/logo.png"
+          alt="MATLAB logo"
+          class="d-block mx-auto mb-3"
+          width="100"
+          height="100" />
+        <h2>Generate a GitHub Actions workflow for MATLAB</h2>
+        <p class="lead text-muted">
+          Enter your repository to generate a starter GitHub Actions workflow
+          file — ready to commit with a click.
+        </p>
+      </div>
+
+      <div class="card shadow-sm rounded-4 main-card">
+        <div class="card-body p-4">
+          <form id="generateForm" class="needs-validation" novalidate>
+            <div class="mb-3">
+              <label for="repo" class="form-label">
+                GitHub username and repository or URL
+                <span class="text-danger">*</span>
+              </label>
+              <input
+                type="text"
+                class="form-control"
+                id="repo"
+                required
+                placeholder="username/repository or URL"
+                autocomplete="off" />
+              <div class="invalid-feedback">
+                Please enter a valid GitHub username and repository or URL
```

Comment:
Error handling should be improved. When I enter an SSH URL, I am provided with this result:

![image](https://github.com/user-attachments/assets/f7447c03-c9d3-4aa1-a2f8-0f327bbc48b5)

Reply:
Do you mind if we handle this in a follow up? I would like to just get this initial version in so we can bash it later today.